test/DynamoCoreTests/Engine/EngineControllerTest.cs

```diff
@@ -0,0 +1,157 @@
+
+        }
+
+        /// <summary>
+        /// This test method will execute the next method from the EngineController class:
+        /// PreviewGraphSyncData(IEnumerable<NodeModel> updatedNodes, bool verboseLogging)
+        /// </summary>
+
+        [Test]
+        [Category("UnitTests")]
+        public void EngineControllerPreviewGraphSyncDataTest()
+        {
+            //Arrange
+            var openPath = Path.Combine(TestDirectory, @"core\nodeLocationTest.dyn");
+            RunModel(openPath);
+            //I created a list with the first node from the workspace
+            List<NodeModel> nodesList = new List<NodeModel>()
+            { CurrentDynamoModel.CurrentWorkspace.Nodes.First() };
+
+            //Act
+            var syncData = CurrentDynamoModel.EngineController.PreviewGraphSyncData(nodesList, false);
+
+            //There is a code section in PreviewGraphSyncData method where the only way to be reached is by passing updatedNodes = null
+            var syncData2 = CurrentDynamoModel.EngineController.PreviewGraphSyncData(null, false);
+
+            //Assert        
+            Assert.IsNotNull(syncData);
+            // Validating that the method PreviewGraphSyncData after passing the parameter updatedNodes = null
+            Assert.IsNull(syncData2);
+        }
+
+        /// <summary>
+        /// This test method will execute the next method from the EngineController class:
+        /// bool GenerateGraphSyncDataForCustomNode(IEnumerable<NodeModel> nodes, CustomNodeDefinition definition, bool verboseLogging)
+        /// </summary>
+
+        [Test]
+        [Category("UnitTests")]
+        public void EngineControllerGenerateGraphSyncDataForCustomNodeTest()
+        {
+            //Arrange
+            CurrentDynamoModel.CurrentWorkspace = null;
+            var openPath = Path.Combine(TestDirectory, @"core\nodeLocationTest.dyn");
+
+            SyncDataManager syncDataManager = new SyncDataManager();
+            syncDataManager.AddNode(Guid.NewGuid(), new IntNode(1));
+
+            GraphSyncData graphSyncdata = syncDataManager.GetSyncData();
+            Queue<GraphSyncData> graphSyncDataQueue = new Queue<GraphSyncData>();
+            graphSyncDataQueue.Enqueue(graphSyncdata);
+
+            //Act
+            //Using reflection we need to set up the Queue field (graphSyncDataQueue) because is private and all the times is empty (all the elements are deleted)
+            FieldInfo field = CurrentDynamoModel.EngineController.GetType().GetField("graphSyncDataQueue", BindingFlags.Instance | BindingFlags.NonPublic);
+            field.SetValue(CurrentDynamoModel.EngineController, graphSyncDataQueue);
+
+            //Assert
+            //The GenerateGraphSyncDataForCustomNodeTest method will raise an exception due that the graphSyncDataQueue has one element
+            Assert.Throws<InvalidOperationException>(() => CurrentDynamoModel.CustomNodeManager.CreateCustomNode("someNode", "someCategory", ""));
```

Comment:
This test looks like we are artificially creating a situation for an exception to occur, not creating a real scenario that could cause it to occur. Is it therefore really useful to have such tests just for the sake of increasing code coverage?

Reply:
@aparajit-pratap Good point, any case we know a crash will be thrown for sure from an existing dyn?

Reply:
@aparajit-pratap 
This is the code section from the GenerateGraphSyncDataForCustomNode( ) method that is reporting without coverage.

              // Any graph updates through the scheduler no longer store their 
                // GraphSyncData in 'graphSyncDataQueue' (any such entry will be 
                // withdrawn from the queue and get associated with an AsyncTask.
                // This check is to ensure that such case does not exist.
                // 
                if (graphSyncDataQueue.Count > 0)
                {
                    throw new InvalidOperationException(
                        "'graphSyncDataQueue' is not empty");
                }
By reading the comments I understood that probably is related to a deprecated code that now always use scheduler for graph updates, then I didn't find any way to execute that code, the graphSyncDataQueue.Count is always 0, then that's why I decided using reflection to populate the Queue in order to reach that section. If you know any other ways (or you know a specific .dyn file) that enable to execute that code please let me know and I can do the changes.

Thanks

Reply:
Not that I know of. I guess we can mark this code to be removed soon; as you said it sounds like it's deprecated. If it's going to be deprecated soon, my vote would be to not add this test.

Reply:
@aparajit-pratap 
Ok, I will discard this test case from this pull request.

Thanks

Reply:
@aparajit-pratap 
This test method was removed from the pull request in the commit : https://github.com/DynamoDS/Dynamo/pull/10705/commits/4d07fccfc603beb0ef3d0140a6307d3bad04df9d

Please review
cc: @QilongTang